### PR TITLE
Add the Status Embed workflow

### LINKED ISSUE #143: Add embed status workflow
Since the Python Discord migration to GH Actions, a [custom workflow](https://github.com/python-discord/bot/blob/main/.github/workflows/status_embed.yaml) is used to webhook status embeds to the `#dev-log` channel:

image

Such embeds provide more information than the defaults that GitHub itself sends:

image

Since #129, the branding repository now runs event validation in Actions, but does not yet use this workflow.

**The aim is therefore to add it to this repository.**

This task should most likely be handled by someone with write access to the repo, as if I understand Actions correctly, workflows added from forks cannot run until merged (meaning the new workflow cannot be tested).

### PR DESCRIPTION
**Pull request by Akarys42**
This PR adds a setup similar to the [`bot` repository](https://pydis.org/bot) to upload the PR details from the Event Validation workflow and use it in a new Status Embed workflow to post a fancy webhook message to `#dev-log`.

Closes #143

### CODE REVIEW

**Review by kwzrd — DISMISSED** [review]
This is probably not particularly valuable, but I would have done the exact same thing. :smile:

File: .github/workflows/status-embed.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    #
+    # 1. We reach the end of the Deploy workflow, without
+    #    it being skipped.
+    #
+    # 2. A `pull_request` triggered a Lint & Test workflow,
+    #    as the sequence always terminates with one run.
+    #
+    # 3. If any workflow ends in failure or was cancelled.
+    if: >-
+      (github.event.workflow_run.name == 'Deploy' && github.event.workflow_run.conclusion != 'skipped') ||
```
Comment (ks129) [inline .github/workflows/status-embed.yml:24]: I don't think we have a deploy job here?

File: .github/workflows/status-embed.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    types:
+      - completed
+
+jobs:
+  status_embed:
+    # We need to send a status embed whenever the workflow
+    # sequence we're running terminates. There are a number
+    # of situations in which that happens:
+    #
+    # 1. We reach the end of the Deploy workflow, without
```
Comment (ks129) [inline .github/workflows/status-embed.yml:16]: Not critical, but maybe the name should be changed here also?

Comment (Akarys42) [reply .github/workflows/status-embed.yml]: Indeedly, fixed in 8d3e82c3dfee4bc2c1f5942cc54a2bb72406edd2, thanks!

**Review by ks129 — APPROVED** [review]

**Pull request merged**